experimental/fluent/func/src/main/java/io/serverlessworkflow/fluent/func/dsl/FuncCallOpenAPIStep.java

```diff
@@ -26,65 +26,93 @@
+
+  public FuncCallOpenAPIStep() {}
+
+  public void setName(String name) {
+    this.name = name;
+  }
+
+  public FuncCallOpenAPIStep document(String uri) {
     steps.add(b -> b.document(uri));
     return this;
   }
 
-  public FuncCallOpenAPISpec document(
+  public FuncCallOpenAPIStep document(
       String uri, AuthenticationConfigurer authenticationConfigurer) {
     steps.add(b -> b.document(uri, authenticationConfigurer));
     return this;
   }
 
-  public FuncCallOpenAPISpec document(URI uri) {
+  public FuncCallOpenAPIStep document(URI uri) {
     steps.add(b -> b.document(uri));
     return this;
   }
 
-  public FuncCallOpenAPISpec document(URI uri, AuthenticationConfigurer authenticationConfigurer) {
+  public FuncCallOpenAPIStep document(URI uri, AuthenticationConfigurer authenticationConfigurer) {
     steps.add(b -> b.document(uri, authenticationConfigurer));
     return this;
   }
 
-  public FuncCallOpenAPISpec operation(String operationId) {
+  public FuncCallOpenAPIStep operation(String operationId) {
     steps.add(b -> b.operation(operationId));
     return this;
   }
 
-  public FuncCallOpenAPISpec parameters(Map<String, Object> params) {
+  public FuncCallOpenAPIStep parameters(Map<String, Object> params) {
     steps.add(b -> b.parameters(params));
     return this;
   }
 
-  public FuncCallOpenAPISpec parameter(String name, String value) {
+  public FuncCallOpenAPIStep parameter(String name, String value) {
     steps.add(b -> b.parameter(name, value));
     return this;
   }
 
-  public FuncCallOpenAPISpec redirect(boolean redirect) {
+  public FuncCallOpenAPIStep redirect(boolean redirect) {
     steps.add(b -> b.redirect(redirect));
     return this;
   }
 
-  public FuncCallOpenAPISpec authentication(AuthenticationConfigurer authenticationConfigurer) {
+  public FuncCallOpenAPIStep authentication(AuthenticationConfigurer authenticationConfigurer) {
     steps.add(b -> b.authentication(authenticationConfigurer));
     return this;
   }
 
-  public FuncCallOpenAPISpec output(OpenAPIArguments.WithOpenAPIOutput output) {
+  public FuncCallOpenAPIStep output(OpenAPIArguments.WithOpenAPIOutput output) {
     steps.add(b -> b.output(output));
     return this;
   }
 
   @Override
-  public void accept(FuncCallOpenAPITaskBuilder builder) {
-    for (var s : steps) {
-      s.accept(builder);
+  protected void configure(
+      FuncTaskItemListBuilder list, Consumer<FuncCallOpenAPITaskBuilder> post) {
+    if (name == null) {
+      list.openapi(
+          builder -> {
+            for (Consumer<CallOpenAPITaskFluent<?>> c : steps) {
+              c.accept(builder); // OpenAPI DSL
+            }
+            post.accept(builder); // when/inputFrom/outputAs/exportAs
+          });
+    } else {
+      list.openapi(
+          name,
+          builder -> {
+            for (Consumer<CallOpenAPITaskFluent<?>> c : steps) {
+              c.accept(builder);
+            }
+            post.accept(builder);
+          });
```

Comment:
You have basically the same if-else construct in two different places, Im wondering if you can do that logic in the openapi-http method, so basically you check for null there, not here